Core/src/Detector/detail/BlueprintHelper.cpp

```diff
@@ -0,0 +1,184 @@
+                         0.5 * (b->boundaryValues[0] + b->boundaryValues[1]);
+                });
+    }
+  }
+
+  // Sort the children
+  if (recursive) {
+    for (auto& child : node.children) {
+      sort(*child, true);
+    }
+  }
+}
+
+void Acts::Experimental::detail::BlueprintHelper::fillGaps(
+    Blueprint::Node& node, bool adjustToParent) {
+  // Return if this is a leaf node
+  if (node.isLeaf()) {
+    return;
+  }
+
+  if (node.boundsType == VolumeBounds::eCylinder) {
+    // Nodes must be sorted
+    sort(node, false);
+
+    // Container values
+    auto cInnerR = node.boundaryValues[0];
+    auto cOuterR = node.boundaryValues[1];
+    auto cHalfZ = node.boundaryValues[2];
+
+    std::vector<std::unique_ptr<Blueprint::Node>> gaps;
+    // Only 1D binning implemented for the moment
+    if (node.binning.size() == 1) {
+      auto bVal = node.binning.front();
+      if (bVal == binZ) {
+        // adjust inner/outer radius
+        if (adjustToParent) {
+          for_each(node.children.begin(), node.children.end(),
+                   [&](auto& child) {
+                     child->boundaryValues[0] = cInnerR;
+                     child->boundaryValues[1] = cOuterR;
+                   });
+        }
+        auto [negC, posC] = cylEndpointsZ(node);
+        // Assume sorted along the local z axis
+        unsigned int igap = 0;
+        for (auto& child : node.children) {
+          auto [neg, pos] = cylEndpointsZ(*child);
+          if ((neg - negC).norm() > s_onSurfaceTolerance) {
+            // Fill a gap node
+            auto gapName = node.name + "_gap_" + std::to_string(igap);
+            auto gapTransform = Transform3::Identity();
+            gapTransform.rotate(node.transform.rotation());
+            gapTransform.translate(0.5 * (neg + negC));
+            auto gap = std::make_unique<Blueprint::Node>(
+                gapName, gapTransform, VolumeBounds::eCylinder,
+                std::vector<ActsScalar>{cInnerR, cOuterR, negC.z() - neg.z()});
+            gaps.push_back(std::move(gap));
+            ++igap;
+          }
+          // Set to new current negative value
+          negC = pos;
+        }
+        // Check if a last one needs to be filled
+        if ((negC - posC).norm() > s_onSurfaceTolerance) {
+          // Fill a gap node
+          auto gapName = node.name + "_gap_" + std::to_string(igap);
+          auto gapTransform = Transform3::Identity();
+          gapTransform.rotate(node.transform.rotation());
+          gapTransform.translate(0.5 * (negC + posC));
+          auto gap = std::make_unique<Blueprint::Node>(
+              gapName, gapTransform, VolumeBounds::eCylinder,
+              std::vector<ActsScalar>{cInnerR, cOuterR, posC.z() - negC.z()});
+          gaps.push_back(std::move(gap));
+        }
+
+      } else if (bVal == binR) {
+        // We have binning in R present
+        if (adjustToParent) {
+          for_each(node.children.begin(), node.children.end(),
```

Comment:
Do you have a `using namespace std;` somewhere? I don't see it.

Reply:
Fixed.